packages/block-editor/src/components/colors/use-colors.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import memoize from 'memize';
+import { startCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Children, cloneElement, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import InspectorControls from '../inspector-controls';
+import PanelColorSettings from '../panel-color-settings';
+import { useBlockEditContext } from '../block-edit';
+
+const createComponent = memoize( ( attribute, color ) => ( { children } ) =>
+	// Clone children, setting the style attribute from the color configuration,
+	// if not already set explicitly through props.
+	Children.map( children, ( child ) =>
+		cloneElement( child, {
+			style: { [ attribute ]: color, ...child.props.style },
```

Comment:
I'd have thought that sometimes it should be classname for the color palette colors. Why it's just style?
Also as mentioned privately, I feel this component adds more complexity that it solves compared to just returning the props that should be applied to any wrapper the block author decides to apply the color for.

Reply:
>I'd have thought that sometimes it should be classname for the color palette colors. Why it's just style?

>>the current HOC breaks validation in a couple of ways. Namely, when settings are not available during validation, #16489, and when settings change.

This is because of using class names. We should avoid it.

>Also as mentioned privately, I feel this component adds more complexity that it solves compared to just returning the props that should be applied to any wrapper the block author decides to apply the color for.

They can still do that by accessing `.color` or `.setColor`. This just saves a lot of code for the most common use case.

Reply:
I'm not sure I understand properly here but using classNames is an important choice we did a long time ago and I don't think we can/should revert to inline styles here. That's one of the main purposes of the color palettes.

Reply:
See #16489

Reply:
That's not a reason to abandon classNames though? we should just make sure they're available right?

Reply:
Why add that extra complexity and implicitly make the block editor depend on the editor which is something we wanted to avoid?

Styles are meant to be used for cases like this where values are dynamically calculated.

Reply:
I provided more context to the pullquote problem in https://github.com/WordPress/gutenberg/pull/16781#discussion_r307965420.